pkg/cloudprovider/provider/hetzner/provider.go

```diff
@@ -376,6 +419,26 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, data *cloudprovidertypes.P
 	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusNotFound {
 		return false, fmt.Errorf("invalid status code returned. expected=%d got=%d", http.StatusOK, res.StatusCode)
 	}
+
+	if instance.(*hetznerServer).server.PlacementGroup != nil {
+		pg, _, err := client.PlacementGroup.Get(ctx, instance.(*hetznerServer).server.PlacementGroup.Name)
+		if err != nil {
+			return false, hzErrorToTerminalError(err, "failed to get placement group")
+		}
+		count := 0
+		for _, s := range pg.Servers {
+			if s != instance.(*hetznerServer).server.ID {
```

Comment:
so many `instance.(*hetznerServer)` :) you can save it into the var btw.

```go
hzServer := instance.(*hetznerServer)
hzServer.this...
hzServer.that...
passItDownTo(hzServer)
```

Reply:
done